src/compiler/program.ts

```diff
@@ -4366,10 +4366,6 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
             }
         }
 
-        if (options.useDefineForClassFields && languageVersion === ScriptTarget.ES3) {

```

Comment:
Do we have a test that still sets the target as es3 and results in option should be removed error.